numpy/tests/test_ctypeslib.py

```diff
@@ -260,3 +260,9 @@ def test_reference_cycles(self):
             b = np.ctypeslib.as_array(newpnt, (N,))
             # now delete both, which should cleanup both objects
             del newpnt, b
+
+    def test_segmentation_fault(self):
+        a = np.ctypeslib.as_ctypes(np.zeros((224,224,3)))
+        # previously segfaulted here due to `as_ctypes` dropping the reference
+        # to the original `np.zeros` object
+        a[0][0][0]
```

Comment:
rather than testing for a segfault, can we explicitly test that the reference is kept?
```python
arr = np.zeros((224, 224, 3))
c_arr = np.ctypeslib.as_ctypes(arr)
arr_ref = weakref.ref(arr)
del arr


# check the reference wasn't cleaned up
assert_(arr_ref() is not None)

# check we avoid the segfault
c_arr[0][0][0]
```

Reply:
I think we're stuck here though - either we test the implementation of the memory allocator, which may or may not throw a segfault on an invalid access - or we test our own implementation of array lifetimes.

If nothing else, it's easier to debug a failing test than a segfaulting one.